packs/opentelemetry_collector/templates/_helpers.tpl

```diff
@@ -0,0 +1,75 @@
+// allow nomad-pack to set the job name
+[[- define "full_job_name" -]]
+[[- if eq .opentelemetry_collector.job_name "" -]]
+[[- .nomad_pack.pack.name | quote -]]
+[[- else -]]
+[[- .opentelemetry_collector.job_name | quote -]]
+[[- end -]]
+[[- end -]]
+
+// only deploys to a region if specified
+[[- define "region" -]]
+[[- if not (eq .opentelemetry_collector.region "") -]]
+region = [[ .opentelemetry_collector.region | quote]]
+[[- end -]]
+[[- end -]]
+
+// generate the vault task config block if enabled
+[[- define "vault_config" -]]
+    [[ if .opentelemetry_collector.vault_config.enabled ]]
+    vault {
+      [[- if .opentelemetry_collector.vault_config.policies ]]
+      policies      = [[ .opentelemetry_collector.vault_config.policies | toPrettyJson ]]
+      [[- end ]]
+      [[- if .opentelemetry_collector.vault_config.change_mode ]]
+      change_mode   = [[ .opentelemetry_collector.vault_config.change_mode ]]
+      [[- end ]]
+      [[- if .opentelemetry_collector.vault_config.change_signal ]]
+      change_signal = [[ .opentelemetry_collector.vault_config.change_signal ]]
+      [[- end ]]
+      [[- if not ( eq .opentelemetry_collector.vault_config.env nil ) ]]
+      env           = [[ .opentelemetry_collector.vault_config.env ]]
+      [[- end ]]
+      [[- if .opentelemetry_collector.vault_config.namespace ]]
+      namespace     = [[ .opentelemetry_collector.vault_config.namespace ]]
+      [[- end ]]
+    }
+    [[ end ]]
+[[- end -]]
+
+// the default map configures the hostmetrics receiver to look at `/hostfs` mounts for system stats
+// it is merged with .task_config.env which will take precedence in the merge
+// in the event of a conflict
+[[- define "env_vars" -]]
+    env {
+      [[- range $key, $value := mergeOverwrite (dict "HOST_PROC" "/hostfs/proc" "HOST_SYS" "/hostfs/sys" "HOST_ETC" "/hostfs/etc" "HOST_VAR" "/hostfs/far" "HOST_RUN" "/hostfs/run" "HOST_DEV" "/hostfs/dev") .opentelemetry_collector.task_config.env -]]
```

Comment:
FWIW: I couldn't figure out how to define a default list or an empty dict to merge into, so I just decided the `HOST_*` vars were fine as defaults.